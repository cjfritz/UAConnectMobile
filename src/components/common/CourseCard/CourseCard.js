import React, { Component } from 'react';
import {
  Card, CardItem, Text, Button, Body, Grid, Row, Col,
} from 'native-base';
import styles from './CourseCard.style';
// common component to render the course card
export default class CourseCard extends Component {
  onEdit = () => {
    const { courseItem, navigation } = this.props;
    navigation.navigate('PlannerEditComponent', { courseItem });
  };

  // render the coures card
  render() {
    // extract props from this.props
    const { courseItem } = this.props;
    const {
      course,
      description,
      term,
      grade,
      units,
    } = courseItem;
    // return what is to be rendered
    return (
      <Card>
        <CardItem style={ styles.headerContainer } header bordered>
          <Text style={ styles.headerText }>
            { course }
          </Text>
        </CardItem>
        <CardItem bordered>
          <Body style={ styles.col }>
            <Text style={ styles.bodyText }>
              { description }
            </Text>
          </Body>
        </CardItem>
        <CardItem bordered>
          <Grid>
            <Col style={ styles.col }>
              <Row style={ styles.row }>
                <Text style={ styles.titleText } header>GPA</Text>
              </Row>
              <Row style={ styles.row }>
                <Text style={ styles.text }>
                  { grade === '' ? 'N/A' : grade }
                </Text>
              </Row>
            </Col>
            <Col style={ styles.col }>
              <Row style={ styles.row }>
                <Text style={ styles.titleText } header>Term</Text>
              </Row>
              <Row style={ styles.row }>
                <Text style={ styles.bodyText }>
                  { term }
                </Text>
              </Row>
            </Col>
            <Col style={ styles.col }>
              <Row style={ styles.row }>
                <Text style={ styles.titleText } header>Units</Text>
              </Row>
              <Row style={ styles.row }>
                <Text style={ styles.bodyText }>
                  { units === '' ? 'N/A' : units }
                </Text>
              </Row>
            </Col>
          </Grid>
        </CardItem>
        <CardItem bordered>
          <Col style={ styles.col }>
            <Row style={ styles.row }>
              <Button style={ styles.headerContainer } onPress={ () => this.onEdit() }>
                <Text style={ styles.headerText }>
                  Edit Course
                </Text>
              </Button>
            </Row>
          </Col>
        </CardItem>
      </Card>
    );
  }
}
