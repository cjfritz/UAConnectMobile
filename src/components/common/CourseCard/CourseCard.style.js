import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  bodyText: {
    alignItems: 'center',
    color: 'black',
    fontSize: 15,
    justifyContent: 'center',
  },
  col: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  headerContainer: {
    alignItems: 'center',
    backgroundColor: 'darkred',
    justifyContent: 'center',
  },
  headerText: {
    color: 'white',
    fontSize: 18,
  },
  row: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  titleText: {
    color: 'darkred',
    fontSize: 18,
  },
});
