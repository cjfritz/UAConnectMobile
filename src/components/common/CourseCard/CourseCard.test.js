import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CourseCard from './CourseCard';



describe('<CourseCard />', () => {
  it('renders with default props', () => {
    const props = {
      courseItem: {},
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<CourseCard { ...props } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });

  it('renders with defined props', () => {
    const props = {
      courseItem: {
        course: 'TestCourse',
        description: 'TestDescription',
        term: 'TestTerm',
        grade: 4.0,
        units: 4,
      },
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<CourseCard { ...props } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });

  it('renders with missing grade', () => {
    const props = {
      courseItem: {
        course: 'TestCourse',
        description: 'TestDescription',
        term: 'TestTerm',
        grade: '',
        units: 4,
      },
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<CourseCard { ...props } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });

  it('renders with missing units', () => {
    const props = {
      courseItem: {
        course: 'TestCourse',
        description: 'TestDescription',
        term: 'TestTerm',
        grade: 4.0,
        units: '',
      },
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<CourseCard { ...props } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });
});

