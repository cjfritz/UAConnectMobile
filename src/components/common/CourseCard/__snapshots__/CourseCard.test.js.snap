// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseCard /> renders with default props 1`] = `
<Styled(Card)>
  <Styled(CardItem)
    bordered={true}
    header={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "darkred",
        "justifyContent": "center",
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "color": "white",
          "fontSize": 18,
        }
      }
    />
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <Styled(Body)
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "alignItems": "center",
            "color": "black",
            "fontSize": 15,
            "justifyContent": "center",
          }
        }
      />
    </Styled(Body)>
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <GridNB>
      <ColumnNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            header={true}
            style={
              Object {
                "color": "darkred",
                "fontSize": 18,
              }
            }
          >
            GPA
          </Styled(Text)>
        </RowNB>
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text) />
        </RowNB>
      </ColumnNB>
      <ColumnNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            header={true}
            style={
              Object {
                "color": "darkred",
                "fontSize": 18,
              }
            }
          >
            Term
          </Styled(Text)>
        </RowNB>
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "alignItems": "center",
                "color": "black",
                "fontSize": 15,
                "justifyContent": "center",
              }
            }
          />
        </RowNB>
      </ColumnNB>
      <ColumnNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            header={true}
            style={
              Object {
                "color": "darkred",
                "fontSize": 18,
              }
            }
          >
            Units
          </Styled(Text)>
        </RowNB>
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "alignItems": "center",
                "color": "black",
                "fontSize": 15,
                "justifyContent": "center",
              }
            }
          />
        </RowNB>
      </ColumnNB>
    </GridNB>
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <ColumnNB
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <RowNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Styled(Button)
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "darkred",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "color": "white",
                "fontSize": 18,
              }
            }
          >
            Edit Course
          </Styled(Text)>
        </Styled(Button)>
      </RowNB>
    </ColumnNB>
  </Styled(CardItem)>
</Styled(Card)>
`;

exports[`<CourseCard /> renders with defined props 1`] = `
<Styled(Card)>
  <Styled(CardItem)
    bordered={true}
    header={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "darkred",
        "justifyContent": "center",
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "color": "white",
          "fontSize": 18,
        }
      }
    >
      TestCourse
    </Styled(Text)>
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <Styled(Body)
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "alignItems": "center",
            "color": "black",
            "fontSize": 15,
            "justifyContent": "center",
          }
        }
      >
        TestDescription
      </Styled(Text)>
    </Styled(Body)>
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <GridNB>
      <ColumnNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            header={true}
            style={
              Object {
                "color": "darkred",
                "fontSize": 18,
              }
            }
          >
            GPA
          </Styled(Text)>
        </RowNB>
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)>
            4
          </Styled(Text)>
        </RowNB>
      </ColumnNB>
      <ColumnNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            header={true}
            style={
              Object {
                "color": "darkred",
                "fontSize": 18,
              }
            }
          >
            Term
          </Styled(Text)>
        </RowNB>
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "alignItems": "center",
                "color": "black",
                "fontSize": 15,
                "justifyContent": "center",
              }
            }
          >
            TestTerm
          </Styled(Text)>
        </RowNB>
      </ColumnNB>
      <ColumnNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            header={true}
            style={
              Object {
                "color": "darkred",
                "fontSize": 18,
              }
            }
          >
            Units
          </Styled(Text)>
        </RowNB>
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "alignItems": "center",
                "color": "black",
                "fontSize": 15,
                "justifyContent": "center",
              }
            }
          >
            4
          </Styled(Text)>
        </RowNB>
      </ColumnNB>
    </GridNB>
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <ColumnNB
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <RowNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Styled(Button)
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "darkred",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "color": "white",
                "fontSize": 18,
              }
            }
          >
            Edit Course
          </Styled(Text)>
        </Styled(Button)>
      </RowNB>
    </ColumnNB>
  </Styled(CardItem)>
</Styled(Card)>
`;

exports[`<CourseCard /> renders with missing grade 1`] = `
<Styled(Card)>
  <Styled(CardItem)
    bordered={true}
    header={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "darkred",
        "justifyContent": "center",
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "color": "white",
          "fontSize": 18,
        }
      }
    >
      TestCourse
    </Styled(Text)>
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <Styled(Body)
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "alignItems": "center",
            "color": "black",
            "fontSize": 15,
            "justifyContent": "center",
          }
        }
      >
        TestDescription
      </Styled(Text)>
    </Styled(Body)>
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <GridNB>
      <ColumnNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            header={true}
            style={
              Object {
                "color": "darkred",
                "fontSize": 18,
              }
            }
          >
            GPA
          </Styled(Text)>
        </RowNB>
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)>
            N/A
          </Styled(Text)>
        </RowNB>
      </ColumnNB>
      <ColumnNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            header={true}
            style={
              Object {
                "color": "darkred",
                "fontSize": 18,
              }
            }
          >
            Term
          </Styled(Text)>
        </RowNB>
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "alignItems": "center",
                "color": "black",
                "fontSize": 15,
                "justifyContent": "center",
              }
            }
          >
            TestTerm
          </Styled(Text)>
        </RowNB>
      </ColumnNB>
      <ColumnNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            header={true}
            style={
              Object {
                "color": "darkred",
                "fontSize": 18,
              }
            }
          >
            Units
          </Styled(Text)>
        </RowNB>
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "alignItems": "center",
                "color": "black",
                "fontSize": 15,
                "justifyContent": "center",
              }
            }
          >
            4
          </Styled(Text)>
        </RowNB>
      </ColumnNB>
    </GridNB>
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <ColumnNB
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <RowNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Styled(Button)
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "darkred",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "color": "white",
                "fontSize": 18,
              }
            }
          >
            Edit Course
          </Styled(Text)>
        </Styled(Button)>
      </RowNB>
    </ColumnNB>
  </Styled(CardItem)>
</Styled(Card)>
`;

exports[`<CourseCard /> renders with missing units 1`] = `
<Styled(Card)>
  <Styled(CardItem)
    bordered={true}
    header={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "darkred",
        "justifyContent": "center",
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "color": "white",
          "fontSize": 18,
        }
      }
    >
      TestCourse
    </Styled(Text)>
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <Styled(Body)
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "alignItems": "center",
            "color": "black",
            "fontSize": 15,
            "justifyContent": "center",
          }
        }
      >
        TestDescription
      </Styled(Text)>
    </Styled(Body)>
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <GridNB>
      <ColumnNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            header={true}
            style={
              Object {
                "color": "darkred",
                "fontSize": 18,
              }
            }
          >
            GPA
          </Styled(Text)>
        </RowNB>
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)>
            4
          </Styled(Text)>
        </RowNB>
      </ColumnNB>
      <ColumnNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            header={true}
            style={
              Object {
                "color": "darkred",
                "fontSize": 18,
              }
            }
          >
            Term
          </Styled(Text)>
        </RowNB>
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "alignItems": "center",
                "color": "black",
                "fontSize": 15,
                "justifyContent": "center",
              }
            }
          >
            TestTerm
          </Styled(Text)>
        </RowNB>
      </ColumnNB>
      <ColumnNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            header={true}
            style={
              Object {
                "color": "darkred",
                "fontSize": 18,
              }
            }
          >
            Units
          </Styled(Text)>
        </RowNB>
        <RowNB
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "alignItems": "center",
                "color": "black",
                "fontSize": 15,
                "justifyContent": "center",
              }
            }
          >
            N/A
          </Styled(Text)>
        </RowNB>
      </ColumnNB>
    </GridNB>
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <ColumnNB
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <RowNB
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Styled(Button)
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "darkred",
              "justifyContent": "center",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "color": "white",
                "fontSize": 18,
              }
            }
          >
            Edit Course
          </Styled(Text)>
        </Styled(Button)>
      </RowNB>
    </ColumnNB>
  </Styled(CardItem)>
</Styled(Card)>
`;
